core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -523,6 +527,7 @@ private void close(Status status, Metadata trailers) {
       cancelListenersShouldBeRemoved = true;
       try {
         closeObserver(observer, status, trailers);
+        channelStats.reportCallEnded(status.isOk());
```

Comment:
Probably worth nesting the try finally.  Exceptions in the users interceptors can cause an exception to be thrown in `closeObserver`